apps/trading/e2e/tests/referrals/test_referrals.py

```diff
@@ -0,0 +1,151 @@
+BUY_ORDERS = [[1, 106], [1, 107], [1, 108]]
+
+
+@pytest.fixture(scope="module")
+def vega(request):
+    with init_vega(request) as vega:
+        yield vega
+
+
+@pytest.fixture(scope="module")
+def continuous_market(vega):
+    market = setup_simple_market(vega, custom_quantum=100000)
+    return setup_continuous_market(vega, market, BUY_ORDERS, SELL_ORDERS, add_liquidity=False)
+
+
+def generate_referrer_expected_value_dic(expected_base_commission, expected_staking_multiplier, expected_final_commission_rate, expected_volume, expected_num_traders, expected_total_commission):
+    return {
+        '[data-testid=my-volume-value]': expected_volume,
+        '[data-testid=total-commission-value]': expected_total_commission,
+        '[data-testid=base-commission-rate-value]': expected_base_commission,
+        '[data-testid=number-of-traders-value]': expected_num_traders,
+        '[data-testid=final-commission-rate-value]': expected_final_commission_rate,
+        '[data-testid=staking-multiplier-value]': expected_staking_multiplier
+    }
+
+
+def generate_referral_expected_value_dic(expected_volume, expected_tier, expected_discount, expected_epochs, expected_epochs_to_next_tier):
+    return {
+        '[data-testid=combined-volume-value]': expected_volume,
+        '[data-testid=current-tier-value]': expected_tier,
+        '[data-testid=discount-value]': expected_discount,
+        '[data-testid=epochs-in-set-value]': expected_epochs,
+        '[data-testid=epochs-to-next-tier-value]': expected_epochs_to_next_tier
+    }
+
+
+def check_tile_values(page: Page, expected_results: dict):
+    if "referrals" in page.url:
+        page.reload()
+    else:
+        page.goto("/#/referrals/")
+
+    for selector, expected_text in expected_results.items():
+        assert selector_contains_text(
+            page, selector, expected_text), f"Expected text '{expected_text}' not found in selector '{selector}'"
+
+
+def create_benefit_tier(minimum_running_notional_taker_volume, minimum_epochs, referral_reward_factor, referral_discount_factor):
+    return {
+        "minimum_running_notional_taker_volume": minimum_running_notional_taker_volume,
+        "minimum_epochs": minimum_epochs,
+        "referral_reward_factor": referral_reward_factor,
+        "referral_discount_factor": referral_discount_factor,
+    }
+
+
+def create_staking_tier(minimum_staked_tokens, referral_reward_multiplier):
+    return {
+        "minimum_staked_tokens": minimum_staked_tokens,
+        "referral_reward_multiplier": referral_reward_multiplier,
+    }
+
+
+def setup_market_and_referral_scheme(vega: VegaService, continuous_market: str, page: Page):
+    page.goto(f"/#/markets/{continuous_market}")
+
+    create_and_faucet_wallet(vega=vega, wallet=PARTY_A)
+    create_and_faucet_wallet(vega=vega, wallet=PARTY_B)
+    forward_time(vega)
+
+    benefit_tiers = []
+    staking_tiers = []
+    for i in range(1, 4):
```

Comment:
I prefer keeping it as the "JSON" as that's how proposals would look when actually created and it makes it super obvious and easier to read.

Reply:
This makes it more scalable. You can make 5000 tiers if you want. You can see what the json would be if you ctrl click into the method

Reply:
I genuinely think it is a bad practice to ctrl c + ctrl v json every time we want to configure referrals. If the schema changes you can just fix in one place this way. If you have one key change, and you have 5 tiers and you use the json in 10 tests, that is 50 places you have to update

Reply:
it's cool, lets go with it.

I'll update my tests as well to match.